src/services/open-api-client.js

```diff
@@ -97,6 +96,25 @@ class OpenApiClient {
     });
   }
 
+  getHost(host, opts) {
+    if (opts.region || opts.edge) {
+      const domain = host.split('.').slice(-2).join('.');
+      const prefix = host.split('.' + domain)[0];
+      let [product, edge, region] = prefix.split('.');
+      if (edge && !region) {
+        region = edge;
+        edge = undefined;
+      }
+      opts.edge = opts.edge || edge;
+      opts.region = opts.region || region || (opts.edge ? 'us1' : undefined);
```

Comment:
nit: This works too: `opts.region = opts.region || region || (opts.edge && 'us1');`